File: benchmarks/cases/mutual/mutual.dart
// Mutual recursion — is_even/is_odd counting evens over a loop.
bool isEven(int n) {
  if (n == 0) {
    return true;
  }
  return isOdd(n - 1);
}

bool isOdd(int n) {
  if (n == 0) {
    return false;
  }
  return isEven(n - 1);
}

void main() {
  int acc = 0;
  for (int i = 1; i < 130000; i++) {
    if (isEven(i % 1000)) {
      acc += 1;
    }
  }
  print(acc);
}

File: benchmarks/cases/mutual/mutual.cs
// Mutual recursion — is_even/is_odd counting evens over a loop.
long acc = 0;
for (long i = 1; i < 130000; i++)
{
    if (IsEven(i % 1000))
    {
        acc += 1;
    }
}
System.Console.WriteLine(acc);

static bool IsEven(long n)
{
    if (n == 0)
    {
        return true;
    }
    return IsOdd(n - 1);
}

static bool IsOdd(long n)
{
    if (n == 0)
    {
        return false;
    }
    return IsEven(n - 1);
}
